cli/plasmo/src/type.ts

```diff
@@ -17,6 +17,7 @@ type GetElement = Getter<Element>
 export type PlasmoCSUIAnchor = {
   element: Element
   type: "overlay" | "inline"
+  inlineInsertPosition?: "beforebegin" | "afterbegin" | "beforeend" | "afterend"
 }
```

Comment:
You should consider using a Discriminating Union here.

Reply:
Sorry if this isn't the ideal format. 

```suggestion
type PlasmoCSUIOverlayAnchor = {
  element: Element
  type: "overlay"
}

type PlasmoCSUIInlineAnchor = {
  element: Element
  type: "inline"
  inlineInsertPosition?: "beforebegin" | "afterbegin" | "beforeend" | "afterend"
}

export type PlasmoCSUIAnchor = PlasmoCSUIOverlayAnchor | PlasmoCSUIInlineAnchor;
```

At this point you could probably also simply name the property `insertPosition`

Reply:
That way when we check `if (anchor.type === "inline") {...}` Typescript will see `inlineInsertPosition`

Here is an example in [TS playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBACgNgQwM4FsD2BhAygVQJIDyAbhAE6IgCCAdgMYAWapUAvFAN4BQUUEcEKCNWAAuKAFF+g4dyihIYgERoS5BCEWcAvp07zo8ZOmz481OAEtqEGgyasOsvgKGiJU17P1Krl65p5fKwgzJDJgGDQkC2ALNGoAfiUAIwgAMyYIVIBzK0UoAB8oRQQ04DIcvMLi1IzSCCEAE3yikrKyJs0dPXADRFRMXDxbRmY2QwGTQlUKEfsiieMhsz8bOlGAbl1aeKRgKBUyWfWmMUXB-GIj9Tmxxx5naTdrAHd3F2EACgBKABovXoiZQzdSKf5aLacHbUPZQILWW5nfpLUzmYK3BxcB4eYRiV7vJ4-f48bzFeEQMGycmhcKRaKxeIpdKZSrUSkQ7a7fZIegIRpoF4ACSi+zY+MkH2AP0h0Nh5T2GLYnwQJ1ISKMF2Gqu+rAAfPcoAB6Q1QHB0NAodDWYRyejQDJwOACqzZKCrKAqxpQAAqWCgLwsjqgOxQYEQVltCH21JhtKiMTi1Cg-IgSGoAHJ9hAAB4WWHxW3QfSyY3B3ZofgAOid2WVqsrMbCpAi8YZ1G+W0CaSgdbspEr+lYLDYinJih1WJ4Hvrj1cDdjzcojQAVghaK4JYSVX356sac26QmCwkEsVSuV6tRmr8oDy+QLhXtvrIdN0gA)

I haven't tested this locally, but it should work.

Reply:
Great feedback btw 👍

Reply:
Thank you, it looks good!